site/ja/tfx/guide/examplegen.md

```diff
@@ -0,0 +1,139 @@
+
+- 入力: CSV や `TFRecord`, BigQuery といった外部のデータソースからのデータ
+- 出力: `tf.Example` レコード
+
+## ExampleGen とその他のコンポーネント
+
+ExampleGen は [TensorFlow Data Validation](tfdv.md) ライブラリを利用するコンポーネントにデータを提供します。
+これには [SchemaGen](schemagen.md), [StatisticsGen](statsgen.md), [Example Validator](exampleval.md) が含まれます。
+また、[TensorFlow Transform](tft.md) を利用する [Transform](transform.md) にもデータを提供し、最終的には推論時にデプロイメントターゲットへとデータを供給します。
+
+## ExampleGen コンポーネントの使い方
+
+サポートされるデータソース (現在、CSV ファイル、TF Example フォーマットの TFRecord ファイル、BigQuery のクエリ結果、の3つがサポートされています) を用いる場合、ExampleGen コンポーネントは典型的には非常に簡単にデプロイ可能で、ほとんど改修を必要とせずに利用できます。
+典型的なコードは次のようになります:
+
+```python
+from tfx.utils.dsl_utils import csv_input
+from tfx.components.example_gen.csv_example_gen.component import CsvExampleGen
+
+examples = csv_input(os.path.join(base_dir, 'data/simple'))
+example_gen = CsvExampleGen(input_base=examples)
+```
+
+次のように、外部の TF Example 形式のファイルを直接読み込むこともできます:
+
+```python
+from tfx.utils.dsl_utils import tfrecord_input
+from tfx.components.example_gen.import_example_gen.component import ImportExampleGen
+
+examples = tfrecord_input(path_to_tfrecord_dir)
+example_gen = ImportExampleGen(input_base=examples)
+```
+
+## カスタム input/output split
+
+Note: この機能は TFX 0.14 以降でのみ利用可能です。
+
+ExampleGen が出力する 学習/評価 データの比率を変更するためには、`output_config` を ExampleGen コンポーネントに設定してください。
+例を次に示します:
+
+```python
+from  tfx.proto import example_gen_pb2
+
+# Input has a single split 'input_dir/*'.
+# Output 2 splits: train:eval=3:1.
+output = example_gen_pb2.Output(
+             split_config=example_gen_pb2.SplitConfig(splits=[
+                 example_gen_pb2.SplitConfig.Split(name='train', hash_buckets=3),
+                 example_gen_pb2.SplitConfig.Split(name='eval', hash_buckets=1)
+             ]))
+examples = csv_input(input_dir)
+example_gen = CsvExampleGen(input_base=examples, output_config=output)
+```
+
+この例の中でどのように `hash_buckets` が設定されているかに注意してください。
+
+すでに入力データが分割されている場合、`input_config` を ExampleGen コンポーネントに設定してください。
+例を次に示します:
+
+```python
+from  tfx.proto import example_gen_pb2
+
+# Input train split is 'input_dir/train/*', eval split is 'input_dir/eval/*'.
+# Output splits are generated one-to-one mapping from input splits.
+input = example_gen_pb2.Input(splits=[
+                example_gen_pb2.Input.Split(name='train', pattern='train/*'),
+                example_gen_pb2.Input.Split(name='eval', pattern='eval/*')
+            ])
+examples = csv_input(input_dir)
+example_gen = CsvExampleGen(input_base=examples, input_config=input)
+```
+
+ファイルベースの ExampleGen コンポーネント (たとえば、 CsvExampleGen や ImportExampleGen) では、`pattern` は入力ファイルをまとめたディレクトリからの相対パスを glob 形式で記述したものになります。
+クエリベースの ExampleGen コンポーネント (たとえば、BigQUeryExampleGen や PrestoExampleGen) では、`pattern` はSQLクエリになります。
+
+デフォルトでは入力データをまとめたディレクトリには単一のファイルがあるものとして扱われます。また、学習/評価用のデータの分割は2:1の割合になるように行われます。
+
+詳細については [proto/example_gen.proto](https://github.com/tensorflow/tfx/blob/master/tfx/proto/example_gen.proto) を参照してください。
+
```

Comment:
[question] If I take a look at the original markdown, there is a section you didn't translate in the PR.

Maybe the original document has been changed after you translated it?

![image](https://user-images.githubusercontent.com/1390726/69046955-77f43f80-0a3d-11ea-8cef-13703a1a704b.png)

https://github.com/tensorflow/tfx/blob/master/docs/guide/examplegen.md

Reply:
Same as above, it was outdated, sorry. Thank you for your review!